plugins/inputs/modbus/modbus_test.go

```diff
@@ -1857,3 +1857,88 @@ func TestConfigurationPerRequestFail(t *testing.T) {
 		})
 	}
 }
+
+func TestRequestGroupingStartingWithOmit(t *testing.T) {
+	modbus := Modbus{
+		Name:              "Test",
+		Controller:        "tcp://localhost:1502",
+		ConfigurationType: "request",
+		Log:               testutil.Logger{},
+	}
+	modbus.Requests = []requestDefinition{
+		{SlaveID: 1,
+			ByteOrder:    "ABCD",
+			RegisterType: "holding",
+			Fields: []requestFieldDefinition{
+				{
+					Name:      "holding-0",
+					Address:   uint16(0),
+					InputType: "INT16",
+					Omit:      true,
+				},
+				{
+					Name:      "holding-1",
+					Address:   uint16(1),
+					InputType: "UINT16",
+					Omit:      true,
+				},
+				{
+					Name:        "holding-2",
+					Address:     uint16(2),
+					InputType:   "INT64",
+					Scale:       1.2,
+					OutputType:  "FLOAT64",
+					Measurement: "modbus",
+				},
+			},
+		},
+	}
+	require.NoError(t, modbus.Init())
+	require.NotEmpty(t, modbus.requests)
+	require.NotNil(t, modbus.requests[1])
+	require.Len(t, modbus.requests[1].holding, 1)
+	require.Equal(t, uint16(2), modbus.requests[1].holding[0].address)
+	require.Equal(t, uint16(4), modbus.requests[1].holding[0].length)
```

Comment:
I think we should not mess with the addresses here (see my overall comment) but rather check that we only scrape out one field from this request
```suggestion
	require.Len(t, modbus.requests[1].holding[0].fields, 1)
```